src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -3032,4 +3032,48 @@ BOOST_AUTO_TEST_CASE(TransposedConvolutionLayerPaddingTest)
   module6.Backward(std::move(input), std::move(output), std::move(delta));
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 0.0);
 }
+
+/**
+ * Test that checks whether the input dimensions are consitent with the input
+ * layer size.
+ */
+BOOST_AUTO_TEST_CASE(NetworkInputSizeTest)
+{
+  // We have to make sure backpropagation through time doesn't take more
+  // time steps than we have.
+  const int maxLineLength = 17;
+  const int hiddenSize = 128;
+  const arma::uword numLetters = 256;
+
+  using MatType = arma::cube;
+  std::vector<std::string> trainingData;
+  trainingData.push_back(std::string("THIS IS THE INPUT."));
+
+  RNN<> rnn(maxLineLength);
+  rnn.Add<IdentityLayer<>>();
+  rnn.Add<LinearNoBias<>>(numLetters / 16, numLetters / 4);
+  rnn.Add<Dropout<>>(0.6);
+  rnn.Add<Linear<>>(hiddenSize / 4, numLetters);
+
+  const auto makeInput = [=](const char *line) -> MatType {
+    const auto strLen = strlen(line);
+    MatType result(numLetters, 1, strLen, arma::fill::zeros);
+    for (size_t i = 0; i < strLen; ++i)
+    {
+      const auto letter = line[i];
+      result.at(static_cast<arma::uword>(letter), 0, i) = 1.0;
+    }
+    return result;
+  };
+
+  std::vector<MatType> inputs(trainingData.size());
+  for (size_t i = 0; i < trainingData.size(); ++i)
+  {
+    inputs[i] = makeInput(trainingData[i].c_str());
+  }
```

Comment:
I think it would be enough if you just initialize `trainingData` without setting any random data in it.
I also deleted some redundancies, because I found it a bit confusing otherwise. An example would be `using MatType = arma::cube;`.
My suggestion would be following:
```
std::string trainingData = "THIS IS THE INPUT."; 
// Alternative: int trainingDataSize = 18;
RNN<> rnn(maxLineLength);
rnn.Add<IdentityLayer<>>();
rnn.Add<LinearNoBias<>>(numLetters / 16, numLetters / 4);
rnn.Add<Dropout<>>(0.6);
rnn.Add<Linear<>>(hiddenSize / 4, numLetters);

std::vector<arma::cube> inputs(trainingData.size());
for (size_t i = 0; i < trainingData.size(); ++i)
{
    inputs[i] = arma::cube(numLetters, 1, trainingData.size(), arma::fill::zeros);
}
```